src/transfers-history/client.ts

```diff
@@ -115,16 +117,31 @@ export class TransfersHistoryClient {
           this.web3Providers[chainId],
           this.eventsQueriers[chainId],
           this.logger,
-          this.transfersRepository,
           depositorAddr
         );
       }
     }
   }
 
   private async getEventsForDepositor(depositorAddr: string) {
-    await Promise.all(Object.values(this.eventsServices[depositorAddr]).map(eventService => eventService.getEvents()));
+    // query all chains to get the events for the depositor address
+    let events = await Promise.all(
+      Object.values(this.eventsServices[depositorAddr]).map(eventService => eventService.getEvents())
+    );
+    const depositEvents = events
+      .flat()
+      .reduce((acc, val) => [...acc, ...val.depositEvents], [] as FundsDepositedEvent[]);
+    const filledRelayEvents = events
+      .flat()
+      .reduce((acc, val) => [...acc, ...val.filledRelayEvents], [] as FilledRelayEvent[]);
+    const blockTimestampMap = events
+      .flat()
+      .reduce((acc, val) => ({ ...acc, ...val.blockTimestampMap }), {} as { [blockNumber: number]: number });
+
+    depositEvents.map(e => this.insertFundsDepositedEvent(e, blockTimestampMap[e.blockNumber]));
+    filledRelayEvents.map(e => this.insertFilledRelayEvent(e));
```

Comment:
Now I insert the events data into the store only after all events are fetched from the chain

Reply:
not quite sure how the problem gets fixed but the changes look good. one small nit is about using map instead of forEach